docs/DeveloperGuide.md

```diff
@@ -1,35 +1,306 @@
 # Developer Guide
 
+## Table of Contents
+
+* Table of Contents
+{:toc}
```

Comment:
Nice use of a Table of Contents, allows me to understand the entire flow and structure well. However, for the 'Product Scope' might be more appropriate to include those details at the start? Allows me to understand who the program targets and what the program can do, giving me more context to understand subsequent diagrams and logic